### PR TITLE
Fix #401: Implement deprecation of the mobile app version

### LINKED ISSUE #401: Implement deprecation of the mobile app version
Backend has to enforce minimal version of the app to be usable for the process.
to ensure the old flawed application cannot be used for enrollment

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/ConfigurationController.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     *
+     * @param request request
+     * @param eciesContext ECIES context.
+     * @return configuration
+     * @throws PowerAuthEncryptionException Thrown when request decryption fails.
+     */
+    @PostMapping
+    @Operation(summary = "Provide enrollment configuration")
+    @PowerAuthEncryption(scope = EciesScope.APPLICATION_SCOPE)
+    public ObjectResponse<ConfigurationResponse> fetchConfiguration(@EncryptedRequestBody Object request,
```
Comment (banterCZ) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/ConfigurationController.java:64]: cc @kober32 
Returns
```json
{
  "status": "OK",
  "responseObject": {
    "mobileApplication": {
      "android": {
        "minimalVersion": "1.4.0",
        "currentVersion": "1.5.4"
      },
      "ios": {
        "minimalVersion": "1.9.0",
        "currentVersion": "2.0.0"
      }
    }
  }
}
```

Comment (kober32) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/ConfigurationController.java:64]: OK with me, @petrdvorak are you okay with such implementation for RB?

> Just a sidenote, for iOS, the values should be `1.5.4` and `2.0.0` to test properly :)

File: enrollment-server-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/enrollment/response/ConfigurationResponse.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @Data
+    public static class MobileApplication {
+
+        private MobileOs iOs;
+
+        private MobileOs android;
+    }
+
+    @Data
+    public static class MobileOs {
```
Comment (romanstrobl) [inline enrollment-server-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/enrollment/response/ConfigurationResponse.java:41]: I am not sure this class represents a mobile OS - it represents the version. Perhaps we can simply call it a `VersionSpecification` or `MobileOsVersion` or something similar.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/MobileApplicationConfigurationProperties.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+@Getter
+public class MobileApplicationConfigurationProperties {
+
+    private final MobileOs iOs = new MobileOs();
+
+    private final MobileOs android = new MobileOs();
+
+    @Getter
+    @Setter
+    public static class MobileOs {
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/MobileApplicationConfigurationProperties.java:43]: dtto

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/ConfigurationController.java (hunk 1 above)
Comment (banterCZ) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/ConfigurationController.java:64]: Ok, I have changed to `ios.minimal-version=1.5.4`

**Review by romanstrobl — APPROVED** [review]
I have just one minor suggestion to improve the naming.

File: enrollment-server-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/enrollment/response/ConfigurationResponse.java (hunk 2 above)
Comment (banterCZ) [reply enrollment-server-api-model/src/main/java/com/wultra/app/enrollmentserver/api/model/enrollment/response/ConfigurationResponse.java:41]: Thanks, `VersionSpecification` sounds good.

**Pull request merged**